### PR TITLE
feat: Allow master data to have no energy suppliers

### PR DESCRIPTION
**Pull request by MWO1024**
## Description

It is a valid and plausible scenario, that a metering point, for a brief period, does not have an attached energy supplier. In this case we must still validate the master data and continue the flow as usual, with the implicit change, that we do not forward the metered data to any energy supplier in this period.

## Co-pilot summary

This pull request includes several changes to the `MeteringPointMasterDataProviderTests` and related classes to improve the handling of energy supplier data and ensure null checks are properly implemented. The most important changes include updating test assertions, modifying methods to handle cases with no energy suppliers, and adding null checks before accessing energy supplier properties.

### Test Improvements:
* Updated assertions in `MeteringPointMasterDataProviderTests` to check for null values and ensure energy supplier data is correctly validated in various test cases. [[1]](diffhunk://#diff-3023eedb6d459bf692a7163cd6bc4643d62f05964d8615d467481326b316c305L63-R65) [[2]](diffhunk://#diff-3023eedb6d459bf692a7163cd6bc4643d62f05964d8615d467481326b316c305L81-R84) [[3]](diffhunk://#diff-3023eedb6d459bf692a7163cd6bc4643d62f05964d8615d467481326b316c305L103-R115) [[4]](diffhunk://#diff-3023eedb6d459bf692a7163cd6bc4643d62f05964d8615d467481326b316c305L133-R163) [[5]](diffhunk://#diff-3023eedb6d459bf692a7163cd6bc4643d62f05964d8615d467481326b316c305L174-R199) [[6]](diffhunk://#diff-3023eedb6d459bf692a7163cd6bc4643d62f05964d8615d467481326b316c305L204-R280) [[7]](diffhunk://#diff-3023eedb6d459bf692a7163cd6bc4643d62f05964d8615d467481326b316c305L278-R346)

### Codebase Enhancements:
* Modified `FlattenMasterDataForParent` and `FlattenMasterDataForChild` methods in `MeteringPointMasterDataProvider` to handle cases with no energy suppliers and return appropriate data structures. [[1]](diffhunk://#diff-3129b674d990c9bcefa2c2fdc518d259f5e64f74e40a54af47301b89b9894b31L151-R175) [[2]](diffhunk://#diff-3129b674d990c9bcefa2c2fdc518d259f5e64f74e40a54af47301b89b9894b31R198-R232) [[3]](diffhunk://#diff-3129b674d990c9bcefa2c2fdc518d259f5e64f74e40a54af47301b89b9894b31L202-R254)
* Added null checks in `GetReceiversFromMasterData` method in `MeteringPointReceiversProvider` to ensure energy supplier data is only accessed when it is present.

## References

https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/613

## Checklist

- [ ] Should the change be behind a feature flag?
- [ ] Can the feature be meaningfully disabled or circumvented if there are issues (e.g., database-breaking changes)?
- [ ] Has it been considered whether data is being delivered to the wrong actor?
- [ ] Subsystem test executed (dev_002/dev_003)
- [ ] Is there time to monitor state of the release to Production?
- [ ] Reference to the task

### CODE REVIEW

File: source/ProcessManager.Orchestrations.Tests/Unit/Processes/BRS_021/ForwardMeteredData/ElectricityMarket/MeteringPointMasterDataProviderTests.cs [hunk 1]
```diff
@@ -171,14 +180,23 @@ public async Task
...
+                    first.EnergySupplier.Should().NotBeNull();
+                    first.EnergySupplier!.Value.Should().Be("1111111111111");
                 },
                 second =>
                 {
                     second.MeteringPointId.Value.Should().Be("faulty-two-master-data-please");
                     second.ValidFrom.Should().Be(new DateTimeOffset(2021, 2, 1, 0, 0, 0, TimeSpan.Zero));
                     second.ValidTo.Should().Be(new DateTimeOffset(2021, 3, 1, 0, 0, 0, TimeSpan.Zero));
-                    second.EnergySupplier.Value.Should().Be("2222222222222");
+                    second.EnergySupplier.Should().NotBeNull();
```
Comment (RasmusGraabaek) [inline source/ProcessManager.Orchestrations.Tests/Unit/Processes/BRS_021/ForwardMeteredData/ElectricityMarket/MeteringPointMasterDataProviderTests.cs:191]: Should we give this test a new name?

**Review by RasmusGraabaek — APPROVED** [review]
